tests/test_s3_utils.py

```diff
@@ -167,3 +189,248 @@ def should_raise_any_client_error_other_than_no_such_key(
+            assert local_etag == s3_etag
+        finally:
+            delete_s3_key(Resource.STORAGE_BUCKET_NAME.resource_name, asset_filename, s3_client)
+
+
+@mark.infrastructure
+def should_return_accurate_etag_value_for_file_smaller_than_single_chunk_size(
+    s3_client: S3Client,
+) -> None:
+    asset_filename = any_safe_filename()
+    asset_contents = any_file_contents(byte_count=S3_DEFAULT_CHUNK_SIZE - 1)
+
+    staging_bucket = Resource.STAGING_BUCKET_NAME.resource_name
+    storage_bucket = Resource.STORAGE_BUCKET_NAME.resource_name
+
+    local_etag = calculate_s3_etag(asset_contents)
+
+    with S3Object(
+        file_object=BytesIO(initial_bytes=asset_contents),
+        bucket_name=staging_bucket,
+        key=f"{asset_filename}",
+    ):
+        try:
+            # import from staging to storage to ensure aws hasn't changed how etag is calculated
+            importer(
+                source_bucket_name=staging_bucket,
+                original_key=asset_filename,
+                target_bucket_name=storage_bucket,
+                new_key=asset_filename,
+                source_s3_client=get_s3_client_for_role(get_s3_role_arn()),
+            )
+
+            s3_etag = get_s3_etag(
+                s3_bucket=storage_bucket, s3_object_key=f"{asset_filename}", logger=get_log()
+            )
+
+            # ensure single chunk size file
+            assert len(local_etag) == SINGLE_CHUNK_ETAG_LENGTH
+            assert local_etag == s3_etag
+        finally:
+            delete_s3_key(Resource.STORAGE_BUCKET_NAME.resource_name, asset_filename, s3_client)
+
+
+@mark.infrastructure
+def should_return_accurate_etag_value_for_single_chunk_size_file(
+    s3_client: S3Client,
+) -> None:
+    asset_filename = any_safe_filename()
+    asset_contents = any_file_contents(byte_count=S3_DEFAULT_CHUNK_SIZE)
+
+    staging_bucket = Resource.STAGING_BUCKET_NAME.resource_name
+    storage_bucket = Resource.STORAGE_BUCKET_NAME.resource_name
+
+    local_etag = calculate_s3_etag(asset_contents)
+
+    with S3Object(
+        file_object=BytesIO(initial_bytes=asset_contents),
+        bucket_name=staging_bucket,
+        key=f"{asset_filename}",
+    ):
+        try:
+            # import from staging to storage to ensure aws hasn't changed how etag is calculated
+            importer(
+                source_bucket_name=staging_bucket,
+                original_key=asset_filename,
+                target_bucket_name=storage_bucket,
+                new_key=asset_filename,
+                source_s3_client=get_s3_client_for_role(get_s3_role_arn()),
+            )
+
+            s3_etag = get_s3_etag(
+                s3_bucket=storage_bucket, s3_object_key=f"{asset_filename}", logger=get_log()
+            )
+
+            # file size exactly at single chunk returns 34 characters etag rather than 32 characters
+            # (i.e. "656dadd6d61e0ebfd29264e34d742df3-1") where -1 suffix denotes 1 chunk
```

Comment:
We might want this at the top of the test to clarify why we have this special case test. Also, "i.e." ("id est" == "that is") should probably be "e.g." ("exempli gratia" == "for example").